Fix #1 in readme

N0tExisting opened the pull request:
Fix #1,
Make the build faster by using `concurrently`, change a small bit of configuration.
Update the workflow to also run on windows and split the jobs.

N0tExisting committed:
Fix Example in readme,
Make build faster &
Small configuration changes

N0tExisting committed:
Update Workflow

Also check on windows and split jobs

N0tExisting committed:
Remove & Shorten Matrices

aldy505 commented on .github/workflows/CI.yml:
```diff
@@ -0,0 +1,127 @@
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
```
these jobs will waste github actions' quota, even if it's free

I would prefer the CI to be just it is now

N0tExisting replied:
Are you talking about the triggers?
This only allows you to run the workflow manually

N0tExisting replied:
If you are talking about the jobs, the repeating parts use almost no quota compared to the rest.
And because we are testing on windows and linux it is quicker to just have one job that runs twice compared to running everything twice.

aldy505 commented on package.json:
```diff
@@ -25,7 +25,7 @@
   "scripts": {
     "dev": "npm run build -- --watch --ignore-watch examples",
-    "build": "npm run build-cjs && npm run build-esm",
+    "build": "concurrently --kill-others-on-fail \"npm:build-*\"",
```
thanks, but sorry I don't intend to add anymore dependency.

making the build time fast is not my goal, we can wait it to build, because it's fast already, it doesn't take any longer than 10 seconds to do so.

N0tExisting replied:
This is a only dev dependency, but sure

aldy505 commented on package.json:
```diff
@@ -53,6 +53,7 @@
     "@typescript-eslint/eslint-plugin": "4.31.1",
     "@typescript-eslint/parser": "4.31.1",
     "c8": "^7.10.0",
+    "concurrently": "^6.4.0",
```
sorry but this must go

aldy505 commented on README.md:
````diff
@@ -42,13 +48,13 @@ module in your application.
-  import routes from 'virtual:generated-pages-solid';
-</script>
+```js
+import { useRoutes } from 'solid-app-router';
+import routes from 'virtual:generated-pages-solid';
+
+const Routes = useRoutes(routes);
 
-<Router {routes} />
+<Routes {routes} />
````
thanks!

N0tExisting committed:
Finish Requested changes

N0tExisting committed:
Fix Eslint Errors

Sorry for adding `solid-js` and`solid-app-router` as DevDependencies, I need types

Also added editorconfig to not have to fight with prettier.

N0tExisting committed:
Make tests also correct on windows (see #2)

N0tExisting committed:
Small stuff that bothered me

N0tExisting committed:
Make actions even shorter

aldy505 commented:
the tests failed

N0tExisting commented:
> the tests failed

Some of it is that the order changed, on windows however its #2

aldy505 commented:
okay gotta fix that soon. merging this now. 

thanks for your contribution!
